content/tutorials/mqtt-snake.textile

```diff
@@ -0,0 +1,539 @@
+
+  ```[python]
+    import paho.mqtt.client as mqtt
+    import curses
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-python-step3
+
+h2. Step 4 - Connect to Ably through MQTT
+
+With the libraries now available, it's time to set up MQTT. When using MQTT with Ably, there are a few requirements with regards to your setup. You'll need to connect to 'mqtt.ably.io' on port 8883, which requires the use of SSL/TLS in your connection. If you have a device which cannot support SSL, you'll need to connect via port 1883 instead, but this will come with a number of "restrictions":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter. 
+
+You will need to set the "keep alive" time value to between 15 and 60 seconds, where 60 seconds will maximize the battery life, and 15 seconds will maximize responsiveness to network issues. For this tutorial you'll be setting it to 15.
+
+For authentication, you'll need to either use an API Key, or a token. For simplicity of the tutorial, you'll be using the API key you obtained in step 1. For this, you'll need to provide the first half of the API Key (before the colon) as the username, and the second half of the API Key (after the colon) as the password. For example, if your API key is @A12B3C.4D:5E6F7G8H@, your username will be @A12B3C.4D@ and your password will be @5E6F7G8H@.
+
+blang[javascript].
+  Add the following code below your current code in @controller.js@, replacing 'FIRST_HALF_OF_API_KEY' and 'SECOND_HALF_OF_API_KEY' with the appropriate parts of your own API Key:
+
+  ```[javascript]
+    var options = { keepAlive: 15,
+      username: 'FIRST_HALF_OF_API_KEY',
+      password: 'SECOND_HALF_OF_API_KEY',
+      port: 8883
+    };
+
+    var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+
+    client.on('connect', function () {
+      console.log('connected!');
+    });
+
+    client.on('error', function(err){
+      console.log(err);
+      client.end();
+    });
+  ```
+
+  With the above code, you set @options@ as described above, and then attempt to connect to Ably's MQTT endpoint. You specify @mqtts@ as part of the host address to specify that the connection attempt should be made with MQTT using SSL/TLS.
+
+  Now, try out the code by typing @node controller.js@ in the command line and you should see in the console that the client successfully connects through MQTT.
+
+  <a href="/images/tutorials/mqtt-snake-connected.png" target="_blank">
+    <img src="/images/tutorials/mqtt-snake-connected.png" style="width: 100%" alt="Connected through MQTT!">
+  </a>
+
+  If you have any issues with this, please check out our "usage notes":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter.
```

Comment:
perhaps instead of "our usage notes", please check out the "Ably MQTT usage notes".